### PR TITLE
Import the ILLinker nuget package

### PR DESCRIPTION
**Pull request by fadimounir**
Import the ILLinker nuget package. ILLink task will be invoked by the SDK.

### CODE REVIEW

**Comment by fadimounir** [comment]
@nguerrera PTAL and merge if it looks good. I don't think i'll have access to merge :)

File: src/redist/targets/BundledSdks.targets [hunk 1]
```diff
@@ -10,5 +10,6 @@
     <BundledSdk Include="Microsoft.NET.Sdk.WindowsDesktop" Version="$(MicrosoftNETSdkWindowsDesktopPackageVersion)" />
     <BundledSdk Include="FSharp.NET.Sdk" Version="1.0.4-bundled-0100" />
     <BundledSdk Include="Microsoft.Docker.Sdk" Version="1.1.0" />
+    <BundledSdk Include="ILLink.Tasks" Version="0.1.6-prerelease.19164.1" />
```
Comment (nguerrera) [inline src/redist/targets/BundledSdks.targets]: This should reference a version in eng/versions.props. And there should probably be automatic dependency flow, which also requires a corresponding entry in version.details.xml.

Comment (fadimounir) [reply src/redist/targets/BundledSdks.targets]: Can you explain to me a bit how the automatic flow would work, and what would the entry in version.details.xml look like? Looking at the existing entries in version.details.xml, they all look hard coded to a specific version/sha, so how do they get updated automatically when some commit gets merged in the various repos?

Comment (nguerrera) [reply src/redist/targets/BundledSdks.targets]: You can leave the `<Sha></Sha>` in the initial commit. What happens is a bot comes and fills in the sha and bumps the version numbers.  See for example #500

Comment (nguerrera) [reply src/redist/targets/BundledSdks.targets]: @sbomer, do you know if the linker repo is publishing to BAR?

Comment (sbomer) [reply src/redist/targets/BundledSdks.targets]: Yup, it is: https://dev.azure.com/dnceng/internal/_build/results?buildId=124335.

Comment (sbomer) [reply src/redist/targets/BundledSdks.targets]: illink.tasks.0.1.6-prerelease.19164.1.nupkg is the version with the Sdk.props changes.

Comment (fadimounir) [reply src/redist/targets/BundledSdks.targets]: Do we have to declared the version as a property in eng/Versions.props, and use that property value in BundledSdks.targets?

Comment (nguerrera) [reply src/redist/targets/BundledSdks.targets]: yes.

Comment (nguerrera) [reply src/redist/targets/BundledSdks.targets]: The updates will edit eng/Versions.props and eng/Version.Details.xml, and nothing else.

Comment (nguerrera) [reply src/redist/targets/BundledSdks.targets]: The property name has to be `NameOfPackageWithoutDotsPackageVersion`. So it should be ILLinkTasksPackageVersion here. Look at the examples more than two lines above.

Comment (nguerrera) [reply src/redist/targets/BundledSdks.targets]: Make a separate property group for the new repo too in Versions.props with a matching comment (follow the pattern). This prevents merge conflicts between updates to different things.

Comment (fadimounir) [reply src/redist/targets/BundledSdks.targets]: Sounds good. I just wasn't sure if the auto updater would also edit Versions.props. Thanks!

**Comment by fadimounir** [comment]
@nguerrera  ping :)

**Pull request merged**